nucypher/cli/deploy.py

```diff
@@ -128,28 +130,46 @@ def deploy(action,
     else:
         deployer_interface = BlockchainInterfaceFactory.get_interface(provider_uri=provider_uri)
 
-    if action == "inspect":
-        if registry_infile:
-            registry = LocalContractRegistry(filepath=registry_infile)
-        else:
-            registry = InMemoryContractRegistry.from_latest_publication()
-        administrator = ContractAdministrator(registry=registry, deployer_address=deployer_address)
-        paint_deployer_contract_inspection(emitter=emitter, administrator=administrator)
-        return  # Exit
+    deployer_interface.connect()
 
     #
     # Establish Registry
     #
 
-    # Establish a contract registry from disk if specified
-    default_registry_filepath = os.path.join(DEFAULT_CONFIG_ROOT, BaseContractRegistry.REGISTRY_NAME)
-    registry_filepath = (registry_outfile or registry_infile) or default_registry_filepath
-    if dev:
+    # FIXME: Revisit all this registry input/output logic - #1329
+    if dev and not registry_infile:
         # TODO: Need a way to detect a geth --dev registry filepath here. (then deprecate the --dev flag)
-        registry_filepath = os.path.join(config_root, 'dev_contract_registry.json')
-    registry = LocalContractRegistry(filepath=registry_filepath)
+        registry_infile = os.path.join(config_root, 'dev_contract_registry.json')
+
+    if action == "contracts":
+        if contract_name and registry_infile is None:  # There's a current deployment and you want to deploy a new contract on it.
+            message = "--registry-infile is required when deploying a contract to an existing registry."
```

Comment:
Adding a command to download the latest registry to disk will help prevent accidents.